connect/runtime/src/test/java/org/apache/kafka/connect/runtime/AbstractHerderTest.java

```diff
@@ -943,35 +943,45 @@ public void testGenerateResultWithConfigValuesWithNoConfigKeysAndWithSomeErrors(
 
     @Test
     public void testSinkConnectorPluginConfig() throws ClassNotFoundException {
-        testConnectorPluginConfig("sink", SampleSinkConnector::new, SampleSinkConnector::config);
+        testConnectorPluginConfig("sink", SampleSinkConnector::new, SampleSinkConnector::config, Optional.of(SinkConnectorConfig.configDef()));
+    }
+
+    @Test
+    public void testSinkConnectorPluginConfigIncludingCommon() throws ClassNotFoundException {
+        testConnectorPluginConfig("sink", SampleSinkConnector::new, SampleSinkConnector::configWithCommon, Optional.empty());
     }
 
     @Test
     public void testSourceConnectorPluginConfig() throws ClassNotFoundException {
-        testConnectorPluginConfig("source", SampleSourceConnector::new, SampleSourceConnector::config);
+        testConnectorPluginConfig("source", SampleSourceConnector::new, SampleSourceConnector::config, Optional.of(SourceConnectorConfig.configDef()));
+    }
+
+    @Test
+    public void testSourceConnectorPluginConfigIncludingCommon() throws ClassNotFoundException {
+        testConnectorPluginConfig("source", SampleSourceConnector::new, SampleSourceConnector::configWithCommon, Optional.empty());
     }
 
     @Test
     public void testConverterPluginConfig() throws ClassNotFoundException {
-        testConnectorPluginConfig("converter", SampleConverterWithHeaders::new, SampleConverterWithHeaders::config);
+        testConnectorPluginConfig("converter", SampleConverterWithHeaders::new, SampleConverterWithHeaders::config, Optional.empty());
     }
 
     @Test
     public void testHeaderConverterPluginConfig() throws ClassNotFoundException {
-        testConnectorPluginConfig("header-converter", SampleHeaderConverter::new, SampleHeaderConverter::config);
+        testConnectorPluginConfig("header-converter", SampleHeaderConverter::new, SampleHeaderConverter::config, Optional.empty());
     }
 
     @Test
     public void testPredicatePluginConfig() throws ClassNotFoundException {
-        testConnectorPluginConfig("predicate", SamplePredicate::new, SamplePredicate::config);
+        testConnectorPluginConfig("predicate", SamplePredicate::new, SamplePredicate::config, Optional.empty());
     }
 
     @Test
     public void testTransformationPluginConfig() throws ClassNotFoundException {
-        testConnectorPluginConfig("transformation", SampleTransformation::new, SampleTransformation::config);
+        testConnectorPluginConfig("transformation", SampleTransformation::new, SampleTransformation::config, Optional.empty());
     }
 
-    private <T> void testConnectorPluginConfig(String pluginName, Supplier<T> newPluginInstance, Function<T, ConfigDef> pluginConfig) throws ClassNotFoundException {
+    private <T> void testConnectorPluginConfig(String pluginName, Supplier<T> newPluginInstance, Function<T, ConfigDef> pluginConfig, Optional<ConfigDef> baseConfig) throws ClassNotFoundException {
```

Comment:
Nit: this line's getting a bit line, can we break it up?
```suggestion
    private <T> void testConnectorPluginConfig(
            String pluginName,
            Supplier<T> newPluginInstance,
            Function<T, ConfigDef> pluginConfig,
            Optional<ConfigDef> baseConfig
    ) throws ClassNotFoundException {
```